moldesign/mathutils/align.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import print_function, absolute_import, division
+from future.builtins import *
+from future import standard_library
+standard_library.install_aliases()
+
+# Copyright 2017 Autodesk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from math import sqrt
+from .. import units as u
+from ..utils import exports
+
+@exports
+def rmsd_align(src_points, dst_points):
+    """ Calculate the 4x4 transformation matrix that aligns a set of source points 
```

Comment:
After looking at the code, I'm going to contradict myself - I think these routines should optionally handle both quantities with units as well as numpy arrays. That means we don't need to strip units manually every time we use this routine. 

Easiest way to do this is probably just strip units out at the beginning and add them at the end. We already do this in several places in the code, but all in a haphazard way (feel free to come up with a cleaner way to do it). Here's the basic idea though.

To strip units at the beginning:
```python
if isinstance(src_points, mdt.units.MdtQuantity):
    units = mdt.units.get_units(src_points)
    p1 = units.value_of(src_points)
    p2 = units.value_of(dst_points)
else:
   units = None
```

To reapply units at the end:
```
if units is not None:
    rmsd_error *= units
```

Reply:
It seems that numpy plays well with pint units, no need to convert the input points. The only modifications needed is to set units and ensure matrix/vector multiply retains the correct units.

Check for units:

    if isinstance(src_points, mdt.units.MdtQuantity):
        units = mdt.units.get_units(src_points)
    else:
        units = 1.0

Multiply martix/vector product by units:

        cdiff = rot_mat.dot(src_points[i]-src_center)*units - (dst_points[i]-dst_center)

Reply:
That works too!